src/ORM/FieldType/DBText.php

```diff
@@ -213,7 +213,7 @@ public function ContextSummary(
         $position = empty($keywords) ? 0 : (int) mb_stripos($text, $keywords);
 
         // We want to search string to be in the middle of our block to give it some context
-        $position = max(0, $position - ($characters / 2));
+        $position = floor(max(0, $position - ($characters / 2)));
```

Comment:
I'm not sure about this one - I think it technically changes the behaviour due to the `if ($position > 0)` checks.
If you can validate that the previous behaviour was not correct then I'm happy to approve this, but otherwise it may be better to `floor` only where it's needed to avoid deprecation warnings.

Reply:
it's fine because `$position' it's used in this context:

`mb_substr($text, 0, $position)`

php 8.1 is complaining about passing in a float instead on an int for the 3rd param e.g. `1.5`

```
php -a
var_dump(mb_substr('abcdef', 0, 3));
string(3) "abc"
php > var_dump(mb_substr('abcdef', 0, 3.5));
string(3) "abc"
```

so using `floor()` gives the exact same result

Reply:
To summarise what we just discussed in person:
This only matters for `$position` = 0.5, in which case it'll be reassigned to 0 anyway here:
https://github.com/silverstripe/silverstripe-framework/blob/f1678781a20b0ae5b6b5ed4821b5dd6b8c38d5dc/src/ORM/FieldType/DBText.php#L218-L224